mantidproject/mantid
# D7YIGPositionCalibration new fitting approach

**Description of work.**

Added the option to choose between fitting the YIG Bragg peaks individually or all of peaks recorded by a single detector simultaneously. Part of this new functionality is an option to **not** perform fitting but provide feedback about the initial guesses to the user by plotting the Gaussian peaks overlaid on the loaded data. 

**To test:**

Run the D7 YIG position calibration using D7 UnitTest data setting `FittingMethod` property to 'None', 'Individual', and 'Global' using the example below. 

```
Load('ILL/D7/396442_396831.nxs', OutputWorkspace='intermediateWavelengthScan')
approximate_wavelength = '4.8' # Angstrom
D7YIGPositionCalibration(InputWorkspace='intermediateWavelengthScan', ApproximateWavelength=approximate_wavelength,
                                         YIGPeaksFile='D7_YIG_peaks.xml', CalibrationOutputFile='test_intermediateWavelength.xml',
                                         MinimalDistanceBetweenPeaks=1.5, BankOffsets="-4,-4,0",
                                         FittingMethod='None', ClearCache=False,
                                         FitOutputWorkspace='intermediateWavelength')
```
1. See whether the overlaid Gaussian peaks guesses are close to peaks in data.

2. With FittingMethod either 'Individual', and 'Global', set the LogLevel to notice to see the output parameters of the calibration: wavelength, and three bank gradients. The wavelength should be close close to 4.86 and the gradients should be consistent with unity down to 1%.



*There is no associated issue.*





---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
